proca/lib/proca/stage/message_v2.ex

```diff
@@ -59,6 +46,38 @@ defmodule Proca.Stage.MessageV2 do
     |> put_action_meta(stage)
   end
 
+  def action_page_data(
+        %ActionPage{
+          locale: locale,
+          name: name,
+          thank_you_template: thank_you_template,
+          supporter_confirm_template: supporter_confirm_template,
+          org: org
+        } = action_page
+      ) do
+    %{
+      "locale" => locale,
+      "name" => name,
+      "thankYouTemplate" => thank_you_template,
+      "thankYouTemplateRef" => MessageV1.action_page_template_ref(action_page),
+      "supporterConfirmTemplate" => supporter_confirm_template || org.supporter_confirm_template
+    }
+  end
+
+  def campaign_data(%Campaign{
+        name: name,
+        title: title,
+        external_id: external_id,
+        contact_schema: contact_schema
+      }) do
+    %{
+      "name" => name,
+      "title" => title,
+      "externalId" => external_id,
+      "contactSchema" => Atom.to_string(contact_schema)
+    }
+  end
+
```

Comment:
Please add tracking part of the action, too. Example "tracking": {
      "campaign": "we.fixthestatusquo.org/",
      "content": "",
      "location": "https://widget.proca.app/d/petition_xyz/de/index.html?proca_test",
      "medium": "we.fixthestatusquo.org",
      "source": "referrer"
  }

Reply:
basically, get it as close as possible (same structure, same elements, same attribute names) as the action messageexport type

```
export type ActionMessageV2 = { 
  actionId: number; 
  actionPageId: number; 
  campaignId: number; 
  orgId: number; 
  org: Organisation; 
  // orgId: number, 
  action: ActionV2; 
  contact: ContactV2; 
  personalInfo: PersonalInfo | null; // null if no encryption 
  campaign: Campaign; 
  actionPage: ActionPage; 
  tracking: Tracking; 
  privacy: PrivacyV2; 
  schema: "proca:action:2"; 
  stage: ProcessStage; 
}; 


export type PrivacyV2 = { 
  withConsent: boolean; // this action had a consent under form (visible or implicit) 
  optIn?: boolean; // consent for communication 
  givenAt?: string; 
  emailStatus: null | "double_opt_in" | "bounce" | "blocked" | "spam" | "unsub"; 
  // email Status describes reliability of email: 
  // - null - no knowledge 
  // - double_opt_in - confirmed that newsletter is wanted 
  // - bounce, blocked - declared by email service that is not deliverable 
  // - unsub, spam - declared by email service that user does not want to be contacted 
  emailStatusChanged: null | string; // JSON date-time format 
}; 


export type ActionV2 = { 
  id?: number, //convenience, copy of actionId set elsewhere in the message 
  actionType: string; 
  customFields: { 
    [key: string]: string | number | boolean | string[] | number[]; 
  }; // map of keys to values, or lists of values, not nested 
  createdAt: string; 
  testing: boolean; // is this a test action? (to be discarded) 
}; 
 
export type Organisation = { 
  name: string, 
  title: string, 
  id?: number, //convenience, copy of actionId set elsewhere in the message 
} 
 
export type Tracking = { 
  source: string; // utm_* 
  medium: string; 
  campaign: string; 
  content: string; 
  location: string; // what url was the form on? 
}; 

 ContactV2 = { 
  email: string; 
  firstName: string; 
  contactRef: string; 
  area: string | null; 
} & { [key: string]: any }; // other keys, usually: 
// lastName, phone, country, postcode, area, 
// address: {region, locality, street, streetNumber} 
// but it can differ for different PII schema 
 
export type Campaign = { 
  title: string; // long name 
  name: string; // technical name 
  externalId: number; // can be set by owner of campaign 
  id?: number, //convenience, copy of campaignId set elsewhere in the message 
}; 
 
export type ActionPage = { 
  locale: string; // language or full locale eg: pl, de_AT 
  name: string; // technical name 
  thankYouTemplate: string; // name of thank you template 
  thankYouTemplateRef: string; // backwards compatibility - id of tempalte resolved from Mailjet etc 
  id?: number, //convenience, copy of actionPageId set elsewhere in the message 
}; 

```

Reply:
@1v4n4 can you update proca-queue eventMessage.ts and push it to share with @destag ?

Reply:
@tttp  fb0a9d134a1c2506fa602bf635438c4c663d244f